File: examples/hello/ViewController.m
#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    Ripe *ripe = [[Ripe alloc] initWithBrand:@"dummy" model:@"dummy"];
    [ripe bindImage:self.imageView];
    [ripe bind:@"post_config" callback:^(NSDictionary *response) {
        [ripe setInitials:@"PT" engraving:@"grey"];
        [ripe setPart:@"piping" material:@"leather_dmy" color:@"brown"];
    }];
    [ripe bind:@"price" callback:^(NSDictionary *price) {
        NSDictionary *total = price[@"total"];
        NSNumber *priceFinal = total[@"price_final"];
        NSString *currency = total[@"currency"];
        [self.labelView setText:[NSString stringWithFormat:@"%@ %@", priceFinal, currency]];
    }];
}

@end

File: examples/hello_swift/ViewController.swift
import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var imageView: UIImageView!
    @IBOutlet weak var labelView: UILabel!

    override func viewDidLoad() {
        super.viewDidLoad()

        let ripe = Ripe(brand: "dummy", model: "dummy")
        ripe.bindImage(imageView)
        _ = ripe.bind("post_config") { config in
            ripe.setInitials("PT", engraving: "grey")
        }
        _ = ripe.bind("price") { (price) in
            let total = price["total"] as? NSDictionary
            let priceFinal = total?["price_final"] as! Double
            let currency = total?["currency"] as! String
            self.labelView.text = "\(priceFinal) \(currency)"
        }
    }


}
